src/peer-store/README.md

```diff
@@ -1,3 +1,120 @@
 # Peerstore
 
-WIP
\ No newline at end of file
+Libp2p's Peerstore is responsible for keeping an updated register with the relevant information of the known peers. It should gather environment changes, be able to take decisions and notice interested parties of relevant changes. The Peerstore comprises four main components: `addressBook`, `keyBook`, `protocolBook` and `metadataBook`. These book components have similar characteristics with the `Javascript Map` implementation.
```

Comment:
I wouldn't mention the Map piece. It's not compliant with a Map and we can't iterate on them, so I would avoid comparing the two. 

> It should gather environment changes, be able to take decisions and notice interested parties of relevant changes.

Can you clarify this? It's sounds like the PeerStore tries to be smart ("decisions"), it should **not** be smart. It should be the single source of truth for all peer data.
- If a subsystem learns about peer data it needs to go directly to the peer store. 
- If any system/application/person wants to know about peers or listen for updates, they need to go to the peerstore.